eclipse-collections-code-generator/src/main/resources/impl/list/immutable/immutablePrimitiveSingletonList.stg

```diff
@@ -173,9 +173,10 @@ final class Immutable<name>SingletonList implements Immutable<name>List, Seriali
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public \<V> ImmutableList\<V> collect(<name>ToObjectFunction\<? extends V> function)
     {
-        return FastList.newListWith(function.valueOf(this.element1)).toImmutable();
+        return (ImmutableList\<V>) FastList.newListWith(function.valueOf(this.element1)).toImmutable();
```

Comment:
This could be simplified with:

`return Lists.immutable.with(function.valueOf(this.element1));
`